frontend/src/screens/studies/participant-controls.tsx

```diff
@@ -0,0 +1,195 @@
+import { React, useState } from '@common'
+import { ParticipantStudy } from '@api'
+import { Box, Menu, Button, Icon, OffCanvas } from '@components'
+import { without, sortBy } from 'lodash-es'
+import { StudySubjectID, StudySubjects, tagOfType } from '@models'
+import slidersIcon from '@iconify-icons/bi/sliders'
+import { useMediaMatch } from 'rooks'
+
+const SortTypes = {
+    'study:type': 'Study Type',
+    'time:low-high': 'Time High to Low',
+    'time:high-low': 'Time Low to High',
+    'points:low-high': 'Points High to Low',
+    'points:high-low': 'Points Low to High',
+}
+
+export interface ControlState {
+    sort?: keyof typeof SortTypes
+    subjects?: StudySubjectID[]
+}
+
+interface ControlProps {
+    state: ControlState
+    onChange(state: ControlState): void
+}
+
+export const Sort:React.FC<ControlProps> = ({ state, onChange }) => {
+    return (
+        <div css={{
+            input: {
+                marginRight: '0.5rem',
+            },
+        }}>
+            {Object.entries(SortTypes).map(([value, label]) => (
+                <label className="dropdown-item" key={value}>
+                    <input
+                        type="radio"
+                        checked={state?.sort == value}
+                        onChange={() => onChange({ ...state, sort: value as any }) }
+                    />
+                    {label}
+                </label>
+            ))}
+        </div>
+    )
+}
+
+
+export const Subjects:React.FC<ControlProps> = ({ state, onChange }) => {
+    const onInputChange = (ev: React.ChangeEvent<HTMLInputElement>) => {
+        if (ev.target.checked) {
+            onChange({
+                ...state,
+                subjects: (
+                    state?.subjects ? state.subjects.concat(ev.target.value as any) : [ev.target.value]
+                ) as any as StudySubjectID[] })
+        } else {
+            onChange({ ...state, subjects: without<StudySubjectID>(state.subjects, ev.target.value as any) })
+        }
+    }
+    return (
+        <div css={{
+            input: {
+                marginRight: '0.5rem',
+            },
+        }}>
+            {Object.entries(StudySubjects).map(([value, label]) => (
+                <label key={value} className="dropdown-item">
+                    <input
+                        type="checkbox"
+                        value={value || ''}
+                        checked={state.subjects?.includes(value as any) || false}
```

Comment:
Isn't "|| false" redundant?

Reply:
hehe, yeah it is, but if it returns undefined and then later false/true, react complains about the input changing from `uncontrolled` to `controlled`.  It's better to consistently return the same type